chore(z2s): use new string encoding for bindings in z2s

tantaman opened the pull request.

grgbkr commented:
I think you need to update compiler-bigint.pg-test.ts as well.

tantaman commented:
> I think you need to update compiler-bigint.pg-test.ts as well.

Thanks. Not sure how I missed that in my `find-all`

grgbkr commented on packages/z2s/src/compiler-bigint.pg-test.ts:
```diff
@@ -254,7 +254,7 @@ describe('compiling ZQL to SQL', () => {
       const query = issueQuery.related('comments');
       const c = compile(ast(query), schema.tables);
-      const sqlQuery = formatPg(c);
+      const sqlQuery = formatPgInternalConvert(c);
```
Would be good to add a test case for a where condition involving a bigint column.

grgbkr reviewed (approved):
LGTM

grgbkr commented on packages/z2s/src/compiler.pg-test.ts:
```diff
@@ -587,7 +593,7 @@ describe('compiling ZQL to SQL', () => {
           q.where('text', 'ILIKE', '%2%').related('author'),
         );
-      const sqlQuery = formatPg(
+      const sqlQuery = formatPgInternalConvert(
```
Would be good to update the comment where filter to be on `createdAt` instead of text so that we exercise a timestamp filter.

tantaman commented:
added tests to compare against bigints and dates. Added `.0` to division of timestamp in order to maintain precision going into and coming out of the DB.


![CleanShot 2025-03-28 at 13 58 47](https://github.com/user-attachments/assets/3387344c-269c-4bd9-9bb9-c655d6f5776d)

grgbkr commented:
> added tests to compare against bigints and dates. Added `.0` to division of timestamp in order to maintain precision going into and coming out of the DB.
> 
> ![CleanShot 2025-03-28 at 13 58 47](https://private-user-images.githubusercontent.com/1009003/428138003-3387344c-269c-4bd9-9bb9-c655d6f5776d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMxODU2MzQsIm5iZiI6MTc0MzE4NTMzNCwicGF0aCI6Ii8xMDA5MDAzLzQyODEzODAwMy0zMzg3MzQ0Yy0yNjljLTRiZDktOWJiOS1jNjU1ZDZmNTc3NmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTAzMjhUMTgwODU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzQ1ZDYyNDFjOWI5YjYwMWQzMWEzYzBkM2MwNzc0NWZmNjQ1OWM2MDNhZGYzYTc4YzEwZjBjYWQ5ZDcxNzkxZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.bUZbzi79U9RaGUE7zA4FGBgctryjzLzmWj4ULWXYL1k)

Good catch.
